tests/test_basic.py

```diff
@@ -64,7 +64,13 @@ def test_sftp_mkdir_rmdir_rename():
     lucky.close()
 
 
-def test_no_recursion_to_console_when_already_gone():
+def test_ftp_session_already_gone():
     lucky = Ska.ftp.SFTP('lucky', logger=logger)
+    # explicitly close paramiko channel, socket, and session
+    lucky.ftp.sock.get_transport().close()
+    lucky.ftp.sock.close()
+    lucky.ftp.close()
+    # delete ftp session
     del lucky.ftp
+    # explicitly run __del__ method
     lucky.__del__()
```

Comment:
I don't really know how to make a clean test for this.  With master (or this branch) this will hang in py.test.  If I do something like

```
sys.setrecursionlimit(10)
```

it won't hang, but I can't seem to find a way to throw / catch an error that it hit the limit.